ChildProject/pipelines/metrics.py

```diff
@@ -145,36 +149,50 @@ def check_callable(row):
         
         self.by = by
         self.period = period
-        
-        #build a dataframe with all the periods we will want for each unit
-        if self.period: 
-            self.periods = pd.interval_range(
-                start=datetime.datetime(1900, 1, 1, 0, 0, 0, 0),
-                end=datetime.datetime(1900, 1, 2, 0, 0, 0, 0),
-                freq=self.period,
-                closed="both",
-            )
-            self.periods= pd.DataFrame(self.periods.to_tuples().to_list(),columns=['period_start','period_end'])
-        
-        self.segments = pd.DataFrame()
-
+        self.segments = segments   
         self.recordings = Pipeline.recordings_from_list(recordings)
         
-        if from_time:
-            try:
-                self.from_time = datetime.datetime.strptime(from_time, "%H:%M:%S")
-            except:
-                raise ValueError(f"invalid value for from_time ('{from_time}'); should have HH:MM:SS format instead")
+        if segments is not None:
+            assert by == 'segments' and period is None and recordings is None and from_time is None and to_time is None, "the <segments> option can not be used with options [period,recordings,from_time,to_time], and <by> should be set to 'segments'"
```

Comment:
perhaps instead tell user:
"the <segments> option cannot be combined with options [period,recordings,from_time,to_time], ...."